contributor-dashboard-legacy/src/pages/SizeComparison.tsx

```diff
@@ -49,47 +23,25 @@ interface SizeSnapshot {
 	[bundleId: string]: { parsed: number; gzip: number };
 }
 
-async function fetchSizeSnapshotAzure(buildId: number): Promise<SizeSnapshot> {
-	const response = await fetch(
-		`https://dev.azure.com/mui-org/material-ui/_apis/build/builds/${buildId}/artifacts?api-version=5.1`
-	);
-
-	const body: AzureApiBody<AzureBuildArtifact[]> = await response.json();
-
-	if (response.status === 200) {
-		const artifacts = body.value;
-		const artifact = artifacts.find(
-			(artifact) => artifact.name === "size-snapshot"
-		);
-
-		const downloadUrl = new URL(artifact!.resource.downloadUrl);
-		downloadUrl.searchParams.set("format", "file");
-		downloadUrl.searchParams.set("subPath", "/size-snapshot.json");
-		return downloadSnapshot("size-snapshot-azure", downloadUrl.toString());
-	}
-
-	throw new Error(`${body.typeKey}: ${body.message}`);
-}
-
 async function fetchSizeSnapshotCircleCI(
-	buildNumber: number
+	buildNumber: number,
 ): Promise<SizeSnapshot> {
 	const response = await fetch(
-		`https://circleci.com/api/v2/project/gh/mui/material-ui/${buildNumber}/artifacts`
+		`/.netlify/functions/circle-ci-artifacts?buildNumber=${encodeURIComponent(buildNumber)}`,
 	);
 	const body: CircleCIApiArtifacts = await response.json();
 
 	if (response.status === 200) {
 		const artifacts = body.items;
-		const artifact = artifacts.find(
-			(artifact) => artifact.path === "size-snapshot.json"
+		const sizeSnapshotArtifact = artifacts.find(
+			(artifact) => artifact.path === "size-snapshot.json",
 		);
 
 		const downloadURL = new URL(
 			"/.netlify/functions/test-profile-artifact",
-			document.baseURI
+			document.baseURI,
 		);
-		downloadURL.searchParams.set("url", artifact!.url);
+		downloadURL.searchParams.set("url", sizeSnapshotArtifact!.url);
```

Comment:
Unless we want to shave off every byte possible from the bundle, I'd throw a nice error instead of the cryptic "Cannot read properties of undefined (reading 'url')".